database/executiveQuery.go

```diff
@@ -30,3 +30,22 @@ func GetAllExecutives() ([]models.Executive, error) {
 	}
 	return execs, nil
 }
+
+// Get all executives in a specific department from database
+func GetAllDepartmentExecutives(department string) ([]models.Executive, error) {
+	var execs []models.Executive
+	query := client.Collection("executive").Where("department", "==", department)
+	iter := query.Documents(context.Background())
+	for doc, err := iter.Next(); err != iterator.Done; doc, err = iter.Next() {
+		if err != nil {
+			return []models.Executive{}, err
+		}
+
+		var tempExec models.Executive
+		if err := doc.DataTo(&tempExec); err != nil {
+			return []models.Executive{}, err
+		}
+		execs = append(execs, tempExec)
+	}
+	return execs, nil
```

Comment:
@IcedTea2K this nil at the end of return might be what is causing the JSON response to show null at the end.

Reply:
Yeah, I'll take a look. The `nil` usually just a away for go to raise error. I usually return `nil` at the end to indicate that the function runs successfully. I'll have to check the handler again how's it handling no error.